pom.xml

```diff
@@ -770,4 +822,78 @@
         </plugins>
     </build>
 
+    <repositories>
+        <repository>
+            <id>Maven Central</id>
+            <name>Maven Repository</name>
+            <url>https://repo.maven.apache.org/maven2</url>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
+
+    <profiles>
+        <profile>
+            <id>release</id>
+            <activation>
+                <property>
+                    <name>deployArtifacts</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-source-plugin</artifactId>
+                        <version>${maven-source-plugin.version}</version>
+                        <executions>
+                            <execution>
+                                <id>attach-sources</id>
+                                <goals>
+                                    <goal>jar-no-fork</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-javadoc-plugin</artifactId>
+                        <version>${maven-javadoc-plugin.version}</version>
+                        <executions>
+                            <execution>
+                                <id>attach-javadocs</id>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <doclint>none</doclint>
+                            <source>11</source>
```

Comment:
We're generating the jars with a java 8 target. Does this need to be 8?

Reply:
The plugin was failing to generate docs for few classes. 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.8.0:jar (attach-javadocs) on project incubator-xtable-api: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/extractor/ExtractFromSource.java:38: error: method references are not supported in -source 7
[ERROR]     Iterator<TableChange> tableChangeIterator = commitsBacklog.getCommitsToProcess().stream().map(conversionSource::getTableChangeForCommit).iterator();
[ERROR]                                                                                                                     ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/sync/TableFormatSync.java:64: error: lambda expressions are not supported in -source 7
[ERROR]         results.put(conversionTarget.getTableFormat(), getSyncResult(conversionTarget, SyncMode.FULL, internalTable, target -> target.syncFilesForSnapshot(snapshot.getPartitionedDataFiles()), startTime, snapshot.getPendingCommits()));
[ERROR]                                                                                                                             ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/sync/TableFormatSync.java:88: error: method references are not supported in -source 7
[ERROR]       }).map(Map.Entry::getKey).collect(Collectors.toList());
[ERROR]                         ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/DataFilesDiff.java:39: error: method references are not supported in -source 7
[ERROR]     Map<String, InternalDataFile> targetPaths = target.stream().collect(Collectors.toMap(InternalDataFile::getPhysicalPath, Function.identity()));
[ERROR]                                                                                                            ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/FilesDiff.java:59: error: lambda expressions are not supported in -source 7
[ERROR]     latestFiles.forEach((key, value) -> {
[ERROR]                                      ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/FilesDiff.java:81: error: method references are not supported in -source 7
[ERROR]     Map<String, InternalDataFile> latestFiles = latestFileGroups.stream().flatMap(group -> group.getFiles().stream()).collect(Collectors.toMap(InternalDataFile::getPhysicalPath, Function.identity()));
[ERROR]                                                                                                                                                                  ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/PartitionFileGroup.java:39: error: method references are not supported in -source 7
[ERROR]     Map<List<PartitionValue>, List<InternalDataFile>> filesGrouped = files.collect(Collectors.groupingBy(InternalDataFile::getPartitionValues));
[ERROR]                                                                                                                            ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/PartitionFileGroup.java:40: error: lambda expressions are not supported in -source 7
[ERROR]     return filesGrouped.entrySet().stream().map(entry -> PartitionFileGroup.builder().partitionValues(entry.getKey()).files(entry.getValue()).build()).collect(Collectors.toList());
[ERROR]                                                       ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] Command line was: /opt/homebrew/Cellar/openjdk@11/11.0.22/libexec/openjdk.jdk/Contents/Home/bin/javadoc -J-Duser.language= -J-Duser.country= @options @argfile
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/apidocs' dir.

```

Reply:
Did you try with 1.8 or 8? Sometimes these tools require 1.8

Reply:
1.8 works.

Reply:
Verified it through the mvn deploy command.